packages/react/src/components/subNavigation/SubNavigation.tsx

```diff
@@ -71,22 +77,49 @@ export const SubNavigation: FunctionComponent<ISubNavigationProps & HTMLAttribut
     };
     const selectedItem = selected || defaultSelected;
 
-    const navItems = map(items, ({id, link, label, disabled}: ISubNavigationItem) => (
-        <li
-            key={id}
-            className={classNames('sub-navigation-item', {
-                'mod-selected': id === selectedItem,
-            })}
-        >
-            <a
-                href={link}
-                className={classNames('sub-navigation-item-link', {disabled})}
-                onClick={(e: MouseEvent<HTMLAnchorElement>) => handleItemClick(e, id)}
-            >
-                {label}
-            </a>
-        </li>
-    ));
+    const navItemsContainDescription = items.some((item) => item.description);
+    const navItems = map(items, ({id, link, label, disabled, description}: ISubNavigationItem) => {
+        const openDescription = description && selectedItem === id;
+
+        if (navItemsContainDescription) {
+            return (
+                <li key={id} className={classNames('sub-navigation-item', {'mod-selected': id === selectedItem})}>
+                    <a
+                        href={link}
+                        className={classNames('sub-navigation-item-link-with-description', {disabled})}
+                        onClick={(e: MouseEvent<HTMLAnchorElement>) => handleItemClick(e, id)}
+                    >
+                        {typeof label === 'string'
+                            ? createElement(
+                                  'div',
+                                  {
+                                      className: 'sub-navigation-item-link-with-description-label',
+                                  },
+                                  `${label}`
+                              )
+                            : label}
+                    </a>
+                    {description && (
+                        <SlideY in={openDescription}>
+                            <div className="sub-navigation-item-description body-m-book-subdued">{description}</div>
+                        </SlideY>
+                    )}
+                </li>
+            );
+        } else {
+            return (
+                <li key={id} className={classNames('sub-navigation-item', {'mod-selected': id === selectedItem})}>
+                    <a
+                        href={link}
+                        className={classNames('sub-navigation-item-link', {disabled})}
+                        onClick={(e: MouseEvent<HTMLAnchorElement>) => handleItemClick(e, id)}
+                    >
+                        {label}
+                    </a>
+                </li>
+            );
+        }
```

Comment:
The `else` is not needed in your situation

```suggestion
        }
            return (
                <li key={id} className={classNames('sub-navigation-item', {'mod-selected': id === selectedItem})}>
                    <a
                        href={link}
                        className={classNames('sub-navigation-item-link', {disabled})}
                        onClick={(e: MouseEvent<HTMLAnchorElement>) => handleItemClick(e, id)}
                    >
                        {label}
                    </a>
                </li>
            );
```